lib-persistence-manager/persistencemgr/redis.go

```diff
@@ -16,63 +16,241 @@
 package persistencemgr
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"log"
 	"math/big"
 	"strconv"
 	"strings"
 	"time"
+	"sync/atomic"
+        "unsafe"
 
 	"github.com/ODIM-Project/ODIM/lib-utilities/config"
 	"github.com/ODIM-Project/ODIM/lib-utilities/errors"
+	redisSentinel "github.com/go-redis/redis/v8"
 	"github.com/gomodule/redigo/redis"
 )
 
+var inMemDBConnPool *ConnPool
+var onDiskDBConnPool *ConnPool
+
 const (
 	errorCollectingData string = "error while trying to collect data: "
 	count               int    = 1000
 )
+var ctx = context.Background()
+// DbType is a alias name for int32
+type DbType int32
+
+const (
+	//InMemory - To select in-memory db connection pool
+	InMemory DbType = iota
+	// OnDisk - To select in-disk db connection pool
+	OnDisk
+)
+
+//RedisExternalCalls containes the methods to make calls to external client libraries of Redis DB
+type RedisExternalCalls interface {
+	newSentinelClient(opt *redisSentinel.Options) *redisSentinel.SentinelClient
+	getMasterAddrByName(ctx context.Context, mset string, snlClient *redisSentinel.SentinelClient) *redisSentinel.StringSliceCmd
+}
+
+type redisExtCallsImp struct{}
+
+func (r redisExtCallsImp) newSentinelClient(opt *redisSentinel.Options) *redisSentinel.SentinelClient {
+	return redisSentinel.NewSentinelClient(opt)
+}
+
+func (r redisExtCallsImp) getMasterAddrByName(ctx context.Context, masterSet string, snlClient *redisSentinel.SentinelClient) *redisSentinel.StringSliceCmd {
+	return snlClient.GetMasterAddrByName(ctx, masterSet)
+}
+
+//NewRedisExternalCalls is Constructor for RedisExternalCalls
+func NewRedisExternalCalls() RedisExternalCalls {
+	return &redisExtCallsImp{}
+}
+
+var redisExtCalls RedisExternalCalls
+
+func init() {
+	redisExtCalls = redisExtCallsImp{}
+}
+
+func sentinelNewClient(host string) *redisSentinel.SentinelClient {
+//	sentinelPort := config.Data.DBConf.RedisSentinelPort
+	rdb := redisExtCalls.newSentinelClient(&redisSentinel.Options{
+		Addr:     host + ":" + "26379",
```

Comment:
is this default port? if so make it constant

Reply:
Done